utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -128,17 +151,12 @@ pub enum State {
 		#[structopt(short, long)]
 		snapshot_path: Option<PathBuf>,
 
-		/// The block hash at which to connect.
-		/// Will be latest finalized head if not provided.
-		#[structopt(short, long, multiple = false, parse(try_from_str = parse::hash))]
-		block_at: Option<String>,
-
 		/// The modules to scrape. If empty, entire chain state will be scraped.
 		#[structopt(short, long, require_delimiter = true)]
 		modules: Option<Vec<String>>,
 
 		/// The url to connect to.
-		#[structopt(default_value = "ws://localhost:9944", parse(try_from_str = parse::url))]
+		#[structopt(short, long, default_value = "ws://localhost:9944", parse(try_from_str = parse::url))]
 		url: String,
```

Comment:
A microoptimization that may or may not be worth it: instead of declaring the `url` parameter multiple times with similar options, you could extract that behavior:

```rust
#[derive(structopt::StructOpt)]
struct UrlParam {
  /// The url to connect to.
  #[structopt(short, long, default_value = "ws://localhost:9944", parse(try_from_str = parse::url))]
  url: String,
}
```

Which would then let you reuse that config wherever you wanted as

```rust
   // a field in some StructOpt
  #[structopt(flatten)]
  url: UrlParam,
```

Reply:
Nice, I would not have thought of this.

To expand I why I didn't put it in shared, since at this point it technically is: IMO keeping it as a flag to the state subcommand is more intuitive and will make more sense if we ever get to the point where we can remove it from live

Reply:
Moved the url to a shared param instead of duplicating across the `State` subcommand. I have been on the fence about what is a better UI, but in order to move block fetching prior to the restructuring of `State` it made it easier to move url to shared